examples/move/big_vector/sources/big_vector.move

```diff
@@ -0,0 +1,1274 @@
+        let (val, rm_fix, _) = self.slice_remove(
+            NO_SLICE,
+            0u128,
+            root_id,
+            0u128,
+            NO_SLICE,
+            depth,
+            key,
+        );
+
+        if (rm_fix == RM_FIX_EMPTY) {
+            if (self.depth == 0) {
+                let Slice<E> {
+                    prev: _,
+                    next: _,
+                    keys: _,
+                    vals,
+                } = df::remove(&mut self.id, root_id);
+
+                // SAFETY: The slice is guaranteed to be empty because
+                // it is a leaf and we received the RM_FIX_EMPTY
+                // fix-up.
+                vals.destroy_empty();
+
+                self.root_id = NO_SLICE;
+            } else {
+                let mut root: Slice<u64> = df::remove(&mut self.id, root_id);
+                self.root_id = root.vals.pop_back();
+                self.depth = self.depth - 1;
+            }
+        };
+
+        val
+    }
+
+    /// Remove the elements between `lo` (inclusive) and `hi`
+    /// (exclusive) from `self`.
+    public fun remove_range<E: store + drop>(
+        _self: &mut BigVector<E>,
+        _lo: u128,
+        _hi: u128,
+    ) {
+        abort 0
+    }
+
+    /// Remove elements from `self` at the indices in `keys`,
+    /// returning the associated values.
+    ///
+    /// Aborts if any of the keys are not found.
+    public fun remove_batch<E: store>(
+        _self: &mut BigVector<E>,
+        _keys: vector<u128>,
+    ): vector<E> {
+        abort 0
+    }
+
+    // === SliceRef ===
+
+    /// Find the slice that contains the key-value pair for `key`,
+    /// assuming it exists in the data structure. Returns the
+    /// reference to the slice and the local offset within the slice
+    /// if it exists, aborts with `ENotFound` otherwise.
+    public fun slice_around<E: store>(
+        self: &BigVector<E>,
+        key: u128,
+    ): (SliceRef, u64) {
+        if (self.root_id == NO_SLICE) {
+            abort ENotFound
+        };
+
+        let (ix, leaf, off) = self.find_leaf(key);
+
+        if (off >= leaf.keys.length()) {
+            abort ENotFound
+        } else if (key != leaf.keys[off]) {
+            abort ENotFound
+        };
+
+        (SliceRef { ix }, off)
+    }
+
+    /// Find the slice that contains the key-value pair corresponding
+    /// to the next key in `self` at or after `key`. Returns the
+    /// reference to the slice and the local offset within the slice
+    /// if it exists, or (NO_SLICE, 0), if there is no matching
+    /// key-value pair.
+    public fun slice_following<E: store>(
+        self: &BigVector<E>,
+        key: u128,
+    ): (SliceRef, u64) {
+        if (self.root_id == NO_SLICE) {
+            return (SliceRef { ix: NO_SLICE }, 0)
+        };
+
+        let (ix, leaf, off) = self.find_leaf(key);
+        if (off >= leaf.keys.length()) {
+            (leaf.next(), 0)
```

Comment:
is there a case where leaf.next() doesn't exist? for example if the key is greater than the highest key present in the tree.

Reply:
Indeed, this can return `(SliceRef { ix: NO_SLICE }, 0)`, this is the case mentioned in the doc comment here (additional emphasis added):

> Returns the reference to the slice and the local offset within the slice if it exists, or **(NO_SLICE, 0), if there is no matching key-value pair.**